lib/theme_check/checks/translation_key_exists.rb

```diff
@@ -5,16 +5,41 @@ class TranslationKeyExists < LiquidCheck
     category :translation
     doc docs_url(__FILE__)
 
+    def initialize
+      @schema_locales = nil
+    end
+
+    def on_schema(node)
+      @schema_locales = JSON.parse(node.value.nodelist.join).dig("locales", @theme.default_locale)
+    end
+
     def on_variable(node)
       return unless @theme.default_locale_json&.content&.is_a?(Hash)
-
       return unless node.value.filters.any? { |name, _| name == "t" || name == "translate" }
       return unless (key_node = node.children.first)
       return unless key_node.value.is_a?(String)
 
-      unless key_exists?(key_node.value) || ShopifyLiquid::SystemTranslations.include?(key_node.value)
-        add_offense(
+      if @schema_locales
+        add_as_offense(
+          key_exists?(key_node.value, @theme.default_locale_json.content) || key_exists?(key_node.value, @schema_locales) || ShopifyLiquid::SystemTranslations.include?(key_node.value),
+          "'#{key_node.value}' does not have a matching entry in '#{@theme.default_locale_json.relative_path}' or '#{node.theme_file.relative_path}",
+          node,
+          key_node
+        )
+      else
+        add_as_offense(
+          key_exists?(key_node.value, @theme.default_locale_json.content) || ShopifyLiquid::SystemTranslations.include?(key_node.value),
           "'#{key_node.value}' does not have a matching entry in '#{@theme.default_locale_json.relative_path}'",
+          node,
+          key_node
+        )
+      end
+    end
+
+    def add_as_offense(boolean, message, node, key_node)
+      unless boolean
```

Comment:
I don't think this works. We might traverse the node before the schema is defined. (See comment about unit test below).

I believe the right way to do this would be to collect the translation nodes as we're traversing the document.

Something like 
```
def on_variable(node)
  if ...
  if ...
  @nodes << node
end

def on_end
  @nodes.each do |node|
    unless (key_node = node...)
    unless translation_exists?(key_node.value)
    add_as_offense(
      "'#{key_node.value'" not found in #{search_path}"
    )
  end

  def translation_exists?(key)
    key_exists?(key, @theme...) ||
    key_exists?(key, @schema_locales) ||
    SystemTranslations.include?(key)
  end
end
```

Also, I think we could clean that up a bit. If we do `@schema_locales = {}`, then we can call `key_exists?(..., @schema_locales)` even when there wasn't a schema in the page. And I believe this would simplify a lot of the logic.